wasm/lib/src/module.dart

```diff
@@ -21,6 +21,10 @@ class WasmModule {
   /// Deserialize a module. See [WasmModule.serialize] for more details.
   WasmModule.deserialize(Uint8List data) : this._(data, true);
 
+  /// Asynchronously compile a module.
+  static Future<WasmModule> compileAsync(Uint8List data) async =>
+      Future<WasmModule>(() => WasmModule(data));
```

Comment:
```suggestion
      WasmModule(data);
```

Reply:
The constructor I'm using here defers execution to a later time, but on the same isolate. My idea was that compilation/instantiation might be expensive, so although I can't do the compile on a different isolate, I can at least give the scheduler a chance to prioritize stuff like UI updates. It would be a bit weird to have an async method that can still block for a while.